hazelcast/src/test/java/com/hazelcast/internal/metrics/impl/MetricsServiceTest.java

```diff
@@ -123,6 +124,14 @@ public void setUp() {
 
     @After
     public void tearDown() {
+        // Destroy JMX beans created during testing.
+        MetricsService metricsService = new MetricsService(nodeEngineMock, () -> metricsRegistry);
```

Comment:
I think alternatively we can move out the construction of the service from the test cases into a factory method that sets a shutdown function object field that we can call in the tearDown. This may prevent some other unwanted potential side-effects. With this approach we construct two services, but shutdown only one. For example, the service submits a scheduled task. I think it is fine for now, but it would be safer with the function proposed.

Reply:
Implemented, thanks for the suggestion.